user-interface/0002-WCON-connect_vega_wallet.md

```diff
@@ -0,0 +1,79 @@
+# Connect Vega wallet & select keys
+
+## Connect wallet
+
+When looking to use Vega via a user interface e.g. Dapp (Decentralized web App), I...
+
+- **should** see a link to get a Vega wallet (in case I don't already have one) (<a name="0002-WCON-036" href="#0002-WCON-036">0002-WCON-036</a>)
+- **should** see a link to connect that opens up connection options (<a name="0002-WCON-001" href="#0002-WCON-001">0002-WCON-001</a>)
+- **should** have a way to easily access help should I need to troubleshoot my connection e.g. link to the docs (<a name="0002-WCON-040" href="#0002-WCON-040">0002-WCON-040</a>)
+- **must** select a connection method / wallet type: (<a name="0002-WCON-002" href="#0002-WCON-002">0002-WCON-002</a>)
+- if Rest:
+  - **must** have the option to input a non-default Wallet location (<a name="0002-WCON-003" href="#0002-WCON-003">0002-WCON-003</a>)
+  - **should** warn if the dapp is unable the see a wallet is running at the wallet location  (<a name="0002-WCON-004" href="#0002-WCON-004">0002-WCON-004</a>)
+  - **must** submit attempt to connect to wallet (<a name="0002-WCON-005" href="#0002-WCON-005">0002-WCON-005</a>)
+  - **could** trigger the app to open on the user's machine with a `vegawallet://` prompt <!--(<a name="0002-WCON-006" href="#0002-WCON-006">0002-WCON-006</a>)-->
+  
+  - if the dapp DOES already have a permission with the wallet: **must** see that wallet is connected (<a name="0002-WCON-007" href="#0002-WCON-007">0002-WCON-007</a>) note: if the user want to connect to a different wallet to the one that they were previously connected with, they will have to hit logout.
+    - if the app uses one key at a time: **should** show what key is active (re-use the last active key) (<a name="0002-WCON-008" href="#0002-WCON-008">0002-WCON-008</a>)
+
+  - if the wallet does NOT have an existing permission with the wallet: **must** prompt user to check wallet app to approve the request to connect wallet: See [Connecting to Dapps](0002-WCON-connect_vega_wallet.md#connect-wallet) for what should happen in wallet app (<a name="0002-WCON-009" href="#0002-WCON-009">0002-WCON-009</a>)
+  
+  - if new keys are given permission: **must** show the user the keys have been approved (<a name="0002-WCON-010" href="#0002-WCON-010">0002-WCON-010</a>)
+    - **should** see [public key(s)](7001-DATA-data_display.md#public-keys). (<a name="0002-WCON-037" href="#0002-WCON-037">0002-WCON-037</a>)
+    - **should** see alias(es) (<a name="0002-WCON-011" href="#0002-WCON-011">0002-WCON-011</a>)
+    - **could** see assets on key(s) (<a name="0002-WCON-012" href="#0002-WCON-012">0002-WCON-012</a>)
+    - **would like to** see positions on key(s) <!--(<a name="0002-WCON-013" href="#0002-WCON-013">0002-WCON-013</a>)-->
+    - if the dapp uses one key at a time: **should** prompt me to select key. See [select/switch keys](#select-and-switch-keys). (<a name="0002-WCON-014" href="#0002-WCON-014">0002-WCON-014</a>)
```

Comment:
Perhaps we want a could for eager connection. 

**could**, if a session is active the wallet could eagerly connect

Reply:
added in https://github.com/vegaprotocol/specs/pull/1181/commits/760887658541c490c51fb231f1836bfab0051258